fix: ok-36571,  ok-36570

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced search results with flexible item click actions for a smoother, customizable navigation experience.
	- Updated search links now direct users to a formatted search engine query based on the input.
	- Improved website handling ensures that external links open with the correct details.
	- Revised bookmark and search modal behavior now automatically returns users to the previous screen upon selection.
	- New functionality allows differentiation of web tab types during website handling.
	- Desktop experience improved by automatically switching to the MultiTabBrowser after adding a new browser home tab.
	- New optional callback for item clicks in search results enhances interactivity.
	- Enhanced accessibility of search item identifiers for broader application use.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
fix: update URL handling in SearchResultContent and useSearchModalData hook

- Changed the URL assignment in SearchResultContent to use the item's URL instead of the search value.
- Updated the useSearchModalData hook to generate a Google search URL based on the search value, improving search result handling.

limichange committed:
refactor: remove console log from useSearchModalData hook for cleaner output

limichange committed:
feat: enhance website handling in Discovery module

- Added logging for debugging in SearchResultContent and useWebSiteHandler hooks.
- Updated handleWebSite calls to include shouldPopNavigation for better navigation management.
- Cleaned up dependencies in useWebSiteHandler for improved performance.

limichange committed:
feat: add onItemClick handler to SearchResultContent for improved item interaction

- Introduced onItemClick prop to ISearchResultContentProps to handle item clicks.
- Updated SearchResultContent to invoke onItemClick when an item is pressed, enhancing user interaction.
- Integrated onItemClick in SearchModal to manage navigation upon item selection.

limichange committed:
refactor: streamline website handling in SearchResultContent and useWebSiteHandler

- Removed console log from SearchResultContent for cleaner output.
- Simplified URL and title assignment in useWebSiteHandler to improve clarity and maintainability.
- Adjusted dependencies in SearchResultContent to enhance performance.

limichange committed:
Merge branch 'x' into OK-36571

sidmorizon commented on packages/kit/src/views/Discovery/hooks/useSearchModalData.ts:
```diff
@@ -74,7 +74,7 @@ export function useSearchModalData(searchValue: string) {
             id: ETranslations.explore_search_placeholder,
           })} "${searchValue}"`,
-          url: '',
+          url: `https://www.google.com/search?q=${searchValue}`,
```
encodeURIComponent(searchValue)

sidmorizon committed:
Merge branch 'x' into OK-36571

limichange committed:
Merge branch 'x' into OK-36571

limichange committed:
Merge branch 'x' into OK-36571

limichange committed:
fix: improve URL encoding in search modal and enhance tab navigation

- Updated the search URL in useSearchModalData to use encodeURIComponent for proper encoding of search values.
- Added a console log in useWebSiteHandler for debugging purposes.
- Modified DesktopCustomTabBar to switch to MultiTabBrowser when adding a browser home tab on desktop.

limichange committed:
feat: enhance discovery actions and search handling

- Added a 'type' parameter to the ContextJotaiActionsDiscovery for better tab management.
- Updated the SearchResultContent to use the search value for URLs instead of the item URL.
- Exported SEARCH_ITEM_ID for improved accessibility in useWebSiteHandler.
- Cleaned up the useWebSiteHandler logic to streamline tab handling and navigation.

limichange committed:
Merge branch 'x' into OK-36571

limichange committed:
refactor: simplify useWebSiteHandler by removing unused imports and variables

- Removed unused browser tab actions and related variables from useWebSiteHandler to streamline the code.
- Cleaned up imports for better clarity and maintainability.

limichange committed:
Merge branch 'x' into OK-36571

originalix committed:
Merge branch 'x' into OK-36571

originalix committed:
Merge branch 'x' into OK-36571
